lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -535,6 +536,16 @@ instance LiftHandler ErrEstimateTxFee where
     handler = \case
         ErrEstimateTxFeeNoSuchWallet e -> handler e
         ErrEstimateTxFeeCoinSelection e -> handler e
+        ErrEstimateTxFeeBackend ErrInvalidTxOutAmount ->
+            apiError err403 CannotValidateSelection $ mconcat
+                [ "I can't estimate transaction fee because transaction contains"
+                , " at least one payment output of value 0."
+                ]
+        ErrEstimateTxFeeBackend ErrExceededInpsOrOuts ->
+            apiError err403 CannotValidateSelection $ mconcat
+                [ "I can't estimate transaction fee because transaction has either"
+                , " more than 255 inputs or more than 255 outputs."
```

Comment:
It'd be better to have these numbers (255) encoded within the error type since this is actually configuration / backend-specific. Some like: 

```hs
ErrExceededInpsOrOuts = ErrExceededInpsOrOuts
    { maxInps :: Int
    , maxOuts :: Int
    }
```